Userland/Libraries/LibGUI/DynamicWidgetContainer.cpp

```diff
@@ -0,0 +1,475 @@
+{
+    Optional<size_t> match = DynamicWidgetContainer::s_open_windows.find_first_index(window);
+    if (match.has_value())
+        DynamicWidgetContainer::s_open_windows.remove(match.value());
+}
+
+void DynamicWidgetContainer::mousedown_event(MouseEvent& event)
+{
+    if (event.button() != MouseButton::Primary)
+        return;
+
+    if (m_expand_collapse_button.contains(event.position()) && !is_container_with_individual_order())
+        set_view_state(view_state() == ViewState::Expanded ? ViewState::Collapsed : ViewState::Expanded);
+
+    if (m_detach_widget_button.contains(event.position()))
+        set_view_state(ViewState::Detached);
+}
+
+void DynamicWidgetContainer::mouseup_event(MouseEvent& event)
+{
+    if (event.button() != MouseButton::Primary)
+        return;
+
+    if (m_is_dragging) {
+        // If we dropped the widget outside of ourself, we would like to detach it.
+        if (!m_parent_container.has_value() && !rect().contains(event.position()))
+            set_view_state(ViewState::Detached);
+
+        if (m_parent_container.has_value()) {
+            bool should_move_position = m_parent_container.value()->check_has_move_target(relative_position().translated(m_current_mouse_position), MoveTargetOp::ClearAllTargets);
+
+            if (should_move_position)
+                m_parent_container.value()->swap_widget_positions(this, relative_position().translated(m_current_mouse_position));
+            else
+                set_view_state(ViewState::Detached);
+        }
+
+        m_is_dragging = false;
+        set_override_cursor(Gfx::StandardCursor::Arrow);
+        update();
+    }
+}
+
+void DynamicWidgetContainer::mousemove_event(MouseEvent& event)
+{
+    if (m_hover_over_control_indicator != m_expand_collapse_button.united(m_detach_widget_button).contains(event.position())) {
+        m_hover_over_control_indicator = !m_hover_over_control_indicator;
+        update();
+    }
+    auto changed_cursor = m_is_dragging ? Gfx::StandardCursor::Move : Gfx::StandardCursor::Arrow;
+    if (m_move_widget_knurl.contains(event.position()) && !m_is_dragging)
+        changed_cursor = Gfx::StandardCursor::Hand;
+
+    if (event.buttons() == MouseButton::Primary && !m_is_dragging) {
+        m_is_dragging = true;
+        m_drag_start_location = event.position();
+        changed_cursor = Gfx::StandardCursor::Move;
+    }
+
+    if (m_is_dragging) {
+        m_current_mouse_position = event.position();
+        if (m_parent_container.has_value()) {
+            m_parent_container.value()->check_has_move_target(relative_position().translated(m_current_mouse_position), MoveTargetOp::SetTarget);
+        }
+        update();
+    }
+
+    if (changed_cursor != override_cursor())
+        set_override_cursor(changed_cursor);
+}
+
+void DynamicWidgetContainer::doubleclick_event(MouseEvent& event)
+{
+    if (event.button() != MouseButton::Primary || is_container_with_individual_order())
```

Comment:
Why does a container with individual order not have the double-click functionality? That doesn't make sense from a user perspective.

Reply:
Iv'e enabled the double click again, there was no real reason for that - i just didn't liked the collapsed main section in pixelpaint - but that should not be a general restriction.